evalml/pipelines/utils.py

```diff
@@ -526,6 +535,21 @@ def _make_new_component_name(model_type, component_name, idx=None):
         idx = " " + str(idx) if idx is not None else ""
         return f"{str(model_type)} Pipeline{idx} - {component_name}"
 
+    def _set_cache_data(
+        cached_data, model_family, cached_component_instances, new_component_name, name
+    ):
+        # sets the new cached component dictionary using the cached data and model family information
+        if len(cached_data) and model_family in list(cached_data.keys()):
+            for hashes, component_instances in cached_data[model_family].items():
+                if hashes not in list(cached_component_instances.keys()):
+                    cached_component_instances[hashes] = {}
+                try:
+                    cached_component_instances[hashes][
+                        new_component_name
+                    ] = cached_data[model_family][hashes][name]
+                except KeyError:
```

Comment:
When will there be a key error? That shouldn't happen because the ensemble is made from the pipelines during search so all the hashes should match?

Reply:
Yep! there shouldn't be any keyerrors if the search proceeds successfully! I kept this in the off-chance that a pipeline maybe failed during search, but can remove.

Reply:
What's the current behavior if we run ensembling and one of the pipelines fails? My guess is that we'd use another pipeline from that model family that didn't fail in the ensemble so we would never try to read from the cache of a failed pipeline. 

I think we should preserve the current behavior on `main` in this case. If we need the KeyError for that let's keep it but add a comment and test for that case!

Reply:
@freddyaboulton You're right! It should only be getting the pipelines from the saved `_best_pipeline_info`, which would only be updated upon successful completion of pipelines. I think we should be able to remove it without issue!